src/platform/unix/mod.rs

```diff
@@ -866,6 +866,13 @@ impl Deref for OsIpcSharedMemory {
     }
 }
 
+impl OsIpcSharedMemory {
+    #[inline]
+    pub unsafe fn deref_mut(&mut self) -> &mut [u8] {
```

Comment:
In general is `implementing`  deref / deref_mut sound for shared memory? Since Rust / LLVM doesn't see the other process accessing the memory, wouldn't we be required to use volatile reads and writes to prevent unwanted optimizations?

Reply:
no, hence marked as unsafe, same safety docs also applies to this.

Reply:
This question is slightly different - let me rephrase it:
Lets assume the user upholds all Safety invariants - Is it possible that LLVM still breaks our code, because it e.g. sees that the written values are never read (in the current process) and thus optimizes out some writes (or reads)

Reply:
I think not, because they are still read when sending back.